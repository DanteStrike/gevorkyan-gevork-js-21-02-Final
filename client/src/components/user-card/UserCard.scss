@import "src/assets/variables";

.user-card {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  min-height: 300px;
  padding: 5%;
  background-color: var(--primary-background);
  border: var(--primary-border);
  border-radius: 15px;

  &__title {
    width: 100%;
    font-size: 20px;
    color: var(--primary-font-color);
    text-align: center;
  }

  &__img {
    flex-grow: 1;
    width: 200px;
    margin-bottom: 10%;
    border-radius: 15px;
    object-fit: cover;
  }
}

@media (max-width: $tabletWidth) {
  .user-card {
    margin: 0 auto;
  }
}
