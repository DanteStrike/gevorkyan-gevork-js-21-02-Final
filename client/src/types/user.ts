export type UserTitleType = 'mr' | 'ms' | 'mrs' | 'miss' | 'dr' | '';
export type UserGenderType = 'male' | 'female' | 'other' | '';

export interface IUserPreview {
  id: string;
  title?: UserTitleType;
  firstName: string;
  lastName: string;
  picture?: string;
}

export interface ILocation {
  street: string;
  city: string;
  state: string;
  country: string;
  timezone: string;
}

export interface IUser extends IUserPreview {
  gender: UserGenderType;
  email: string;
  dateOfBirth: string;
  registerDate: string;
  phone: string;
  location: ILocation;
}

export interface IUserRegistration {
  firstName: string;
  lastName: string;
  gender: UserGenderType;
  dateOfBirth: string;
  email: string;
  phone: string;
}

export interface IUserUpdate extends Omit<IUserRegistration, 'email'> {
  id: string;
  picture: string;
}
