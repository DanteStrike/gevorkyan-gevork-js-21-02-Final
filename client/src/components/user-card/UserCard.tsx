import React from 'react';
import './UserCard.scss';
import {Tooltip} from 'antd';
import {IUserPreview} from '../../types';
import {DataUtils} from '../../utils';

interface IUserCardProps {
  user: IUserPreview;
}

function UserCard({user}: IUserCardProps) {
  const {firstName, lastName, title, picture} = user;
  const name = DataUtils.collectFullName(firstName, lastName, title);

  return (
    <article className="user-card" id={`#user-${user.id}`}>
      {picture ? <img className="user-card__img" alt={name} src={picture} /> : <div className="user-card__img" />}
      <Tooltip
        placement="top"
        title={user.id}
        getPopupContainer={() => document.querySelector(`#user-${user.id}`) || document.body}
      >
        <h2 className="user-card__title">{name}</h2>
      </Tooltip>
    </article>
  );
}

export default React.memo(UserCard);
