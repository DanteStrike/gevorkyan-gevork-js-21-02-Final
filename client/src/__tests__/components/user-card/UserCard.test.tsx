import {shallow} from 'enzyme';
import React from 'react';
import UserCard from '../../../components/user-card/UserCard';
import {userOne} from '../../mocks/user.mock';

describe(`Component UserPreview should render correctly`, () => {
  it(`snapshot default test`, () => {
    const wrapper = shallow(<UserCard user={userOne} />);

    expect(wrapper.find(`div.user-card__img`)).toHaveLength(0);
    expect(wrapper.find(`img.user-card__img`)).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it(`snapshot undefined picture test`, () => {
    const wrapper = shallow(<UserCard user={{...userOne, picture: undefined}} />);

    expect(wrapper.find(`div.user-card__img`)).toHaveLength(1);
    expect(wrapper.find(`img.user-card__img`)).toHaveLength(0);
    expect(wrapper).toMatchSnapshot();
  });
});
