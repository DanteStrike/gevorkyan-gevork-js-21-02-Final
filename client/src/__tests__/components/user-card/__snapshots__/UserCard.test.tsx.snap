// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component UserPreview should render correctly snapshot default test 1`] = `
<article
  className="user-card"
  id="#user-userID_1"
>
  <img
    alt="Ms. antLN anyFN"
    className="user-card__img"
    src="url"
  />
  <Tooltip
    arrowPointAtCenter={false}
    autoAdjustOverflow={true}
    getPopupContainer={[Function]}
    mouseEnterDelay={0.1}
    mouseLeaveDelay={0.1}
    placement="top"
    title="userID_1"
  >
    <h2
      className="user-card__title"
    >
      Ms. antLN anyFN
    </h2>
  </Tooltip>
</article>
`;

exports[`Component UserPreview should render correctly snapshot undefined picture test 1`] = `
<article
  className="user-card"
  id="#user-userID_1"
>
  <div
    className="user-card__img"
  />
  <Tooltip
    arrowPointAtCenter={false}
    autoAdjustOverflow={true}
    getPopupContainer={[Function]}
    mouseEnterDelay={0.1}
    mouseLeaveDelay={0.1}
    placement="top"
    title="userID_1"
  >
    <h2
      className="user-card__title"
    >
      Ms. antLN anyFN
    </h2>
  </Tooltip>
</article>
`;
